pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -55,268 +136,511 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	pred := predicate.Funcs{
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			newObj := e.ObjectNew.(*kubevirtv1alpha1.NodeMaintenance)
-			return !newObj.DeletionTimestamp.IsZero()
-		},
-	}
+	return c.Watch(&source.Kind{Type: &corev1.Node{}}, &handler.EnqueueRequestForObject{})
+}
+
+// Reconcile monitors Nodes and creates the MachineRemediation object when the node has reboot annotation.
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 
-	// Create a source for watching noe maintenance events.
-	src := &source.Kind{Type: &kubevirtv1alpha1.NodeMaintenance{}}
+	r.reqLogger = log.WithFields(log.Fields{"Request.Namespace": request.Namespace, "Request.Name": request.Name})
```

Comment:
Do you have an example of the contents of ``request.Name`` and ``request.Namespace``?
I would expect ``request.Namespace`` to contain no useful information since all nodes are in the default namespace.
Does ``request.Name`` match the Node name?  If not, the later would be arguably more useful to know.

Reply:
the namespace you will get here is probably the namespace of the operator itself , request name will most likely have the node maintenance CR spec.name i believe (not sure it will hold the node name)